### ISSUE TITLE
feature request: stop connection as soon as TCP connection breaks

### ISSUE THREAD

**Issue by ilka-schulz** [Zahner-elektrik/Thales-Remote-Python#13]
see #12 for full explanation

**Comment by maxkrapp1** [comment]
When the TCP connection breaks, an exception is thrown.

I caused this error by simply terminating the term in the task manager.

```
Traceback (most recent call last):
  File "C:\x\Thales-Remote-Python\Examples\BasicIntroduction\BasicIntroduction.py", line 35, in <module>
    zahnerZennium.forceThalesIntoRemoteScript()
  File "C:\x\Thales-Remote-Python\thales_remote\script_wrapper.py", line 1618, in forceThalesIntoRemoteScript
    self.remoteConnection.sendStringAndWaitForReplyString(
  File "C:\x\Thales-Remote-Python\thales_remote\connection.py", line 233, in sendStringAndWaitForReplyString
    return self.waitForStringTelegram(answer_message_type, timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\x\Thales-Remote-Python\thales_remote\connection.py", line 213, in waitForStringTelegram
    retval = self.waitForBinaryTelegram(message_type, timeout).decode("ASCII")
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\x\Thales-Remote-Python\thales_remote\connection.py", line 200, in waitForBinaryTelegram
    raise TermConnectionError("Socket error during data reception.")
thales_remote.error.TermConnectionError: Socket error during data reception.
```

Afterwards the connection must be established completely from the beginning again. Including the call of the consturctor, so that the internals are reset.

So after a termination of the connection these steps are necessary.

```python
zenniumConnection = ThalesRemoteConnection()
zenniumConnection.connectToTerm(TARGET_HOST, "ScriptRemote")
```

#12 is gemerged is your feature fully implemented? Or do you have any additions to this?

**Comment by ilka-schulz** [comment]
@maxkrapp1 Thanks for the explanation and the merge!

However, the issue lies in the Thales Term software. If I can not call `disconnectFromTerm()`, the term software goes into an unrecoverable state. It just stays in the "connected" state and it is not possible to get it out of that state or to reconnect by any reasonable means. The only chance is to manually click `NMI & Resethandling` -> `Full Reset` on the ThalesXT GUI. It takes quite some time, it is a manual process and it stop currently running measurements.

Not being able to call `disconnectFromTerm` may happen more often that one might think, e.g. due to network errors, client errors, debugging sessions, etc., especially when using the device for long-term tests. I thus think it is critical for my operations that the Term software continues to accept connections even after a previous connection failed.

**Comment by maxkrapp1** [comment]
Yes this is unfortunately a known problem that it can happen that the Term comes into an unrecoverable state if you do not cleanly disconnect.

If you have a CPU card without a switch, then you can also trigger the reset via software for which a exe is included in the Thales installation.

First exit Thales and then execute the CPU reset.
I use the following command combination for this.
```
taskkill /IM term131.exe /T /F
C:\FLINK\tools\CPUReset\ZenniumReset.exe vcc
```

**Comment by ilka-schulz** [comment]
> ```
> taskkill /IM term131.exe /T /F
> C:\FLINK\tools\CPUReset\ZenniumReset.exe vcc
> ```

Any chance that this bug can be fixed? I can work with the bug for now but this would be quite an issue in the production environment. Alternatively, running Term131 under Linux would work for me, too (that would actually be even better).

**Comment by maxkrapp1** [comment]
I have spoken with the responsible software developer and we will check ways to improve or fix the error behavior. But I can't promise anything here, because the problem is relatively complex.

On the subject of Term under Linx @DF9IR has sent you a mail what he has tested so far.

**Comment by maxkrapp1** [comment]
This was fixed with term version 5.8.8.
